src/main/java/com/ibm/watson/modelmesh/payload/RemotePayloadProcessor.java

```diff
@@ -58,16 +62,34 @@ private static PayloadContent prepareContentBody(Payload payload) {
         ByteBuf byteBuf = payload.getData();
         String data;
         if (byteBuf != null) {
-            ByteBuf encoded = Base64.encode(byteBuf, byteBuf.readerIndex(), byteBuf.readableBytes(), false);
-            //TODO custom jackson serialization for this field to avoid round-tripping to string
-            data = encoded.toString(StandardCharsets.US_ASCII);
+            data = encodeBinaryToString(byteBuf);
         } else {
             data = "";
         }
+        Metadata metadata = payload.getMetadata();
+        Map<String, String> metadataMap = new HashMap<>();
+        if (metadata != null) {
+            for (String key : metadata.keys()) {
+                if (key.endsWith("-bin")) {
+                    byte[] bytes = metadata.get(Metadata.Key.of(key, Metadata.BINARY_BYTE_MARSHALLER));
+                    if (bytes != null) {
```

Comment:
I don't think this null check is needed here

Reply:
pragmatically speaking yes, this is probably not needed as MM doesn't set `null` `Metadata` contents. The reason I  have put that is that (i) the static code analyzer complains about this (ii) for safeguarding eventual future misbehaviours.
please let me know if you prefer that to be removed anyway and I'll proceed.

Reply:
Sure, I'm fine either way. This was only a minor suggestion, since you have just pulled the key from `metadata.keys()` it should always be present and so null here would indicate some bug in this logic. And seemed inconsistent because you're checking in the -bin case but not the ascii case.

Reply:
ok, good point, I'll drop the null check entirely then.